### PR TITLE
INTMDB-676: [CFN-PI] Remove Cloud provider snapshot and Cloud provider snapshot restore jobs

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description
This PR removes `Remove Cloud provider snapshot` and `Cloud provider snapshot restore jobs` for CFN and CDK as they have been removed to `CloudBackupSnapshot` and `CloudBackupRestoreJobs`.

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change works in Atlas

### CODE REVIEW

File: cfn-resources/autogen/schemas/mongodb-atlas-x509authenticationdatabaseuser.json [hunk 1]
```diff
@@ -1,292 +0,0 @@
-{
```
Comment (Zuhairahmed) [inline cfn-resources/autogen/schemas/mongodb-atlas-x509authenticationdatabaseuser.json:1]: Should we also be deleting x509 / other resource schemas as well?

**Review by Zuhairahmed — APPROVED** [review]
minor comment, not blocking. LGTM

File: cfn-resources/autogen/schemas/mongodb-atlas-x509authenticationdatabaseuser.json (hunk 1 above)
Comment (andreaangiolillo) [reply cfn-resources/autogen/schemas/mongodb-atlas-x509authenticationdatabaseuser.json:1]: I don't know which other schemas you are referring to. Could you share a link to these schemas, please? I will look to see if we should remove those as well. Thanks!

**Review by maastha — COMMENTED** [review]
Should we also update [this README ](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/cfn-resources)

**Review by maastha — APPROVED** [review]

File: cfn-resources/autogen/schemas/mongodb-atlas-x509authenticationdatabaseuser.json (hunk 1 above)
Comment (andreaangiolillo) [reply cfn-resources/autogen/schemas/mongodb-atlas-x509authenticationdatabaseuser.json:1]: I will open a follow-up PR to address this comment if needed.

**Comment by andreaangiolillo** [comment]
@maastha no they are not published

**Pull request merged**